### PR TITLE
Optionally avoid expensive collect calls for skew optimized AsOfJoin

### PR DESCRIPTION
**Pull request by souvik-databricks**
Solution for issue https://github.com/databrickslabs/tempo/issues/104

- Addition of a flag to asOfJoin method ignore_null_warning that defaults to False, so users that are well-aware of the risk of nulls can set the flag to True, and tempo can skip the collect calls.

- Checking the logging level for the tempo module, and avoid collect calls when the logging level is above warning. https://docs.python.org/3/howto/logging.html#optimization

### CODE REVIEW

File: python/tempo/tsdf.py [hunk 1]
```diff
@@ -150,10 +150,12 @@ def __getLastRightRow(self, left_ts_col, right_cols, sequence_col, tsPartitionVa
     if tsPartitionVal is not None:
       for column in df.columns:
         if (column.startswith("non_null")):
-          any_blank_vals = (df.agg({column: 'min'}).collect()[0][0] == 0)
-          newCol = column.replace("non_null_ct", "")
-          if any_blank_vals:
-            logger.warning("Column " + newCol + " had no values within the lookback window. Consider using a larger window to avoid missing values. If this is the first record in the data frame, this warning can be ignored.")
+          # Avoid collect() calls when explicitly ignoring the warnings about null values due to lookback window.
+          if not suppress_null_warning and logger.isEnabledFor(logging.WARNING):
+            any_blank_vals = (df.agg({column: 'min'}).collect()[0][0] == 0)
```
Comment (rportilla-databricks) [inline python/tempo/tsdf.py:155]: The original request actually does the collect if the WARNING level is enabled. Switch 'and' to 'or'.

File: python/tempo/tsdf.py [hunk 2]
```diff
@@ -315,6 +317,8 @@ def asofJoin(self, right_tsdf, left_prefix=None, right_prefix="right", tsPartiti
     :param tsPartitionVal - value to break up each partition into time brackets
     :param fraction - overlap fraction
     :param skipNulls - whether to skip nulls when joining in values
+    :param sql_join_opt - if set to True, will use standard Spark SQL join if it is estimated to be efficient
+    :param suppress_null_warning - when tsPartitionVal is specified, will collect min of each column and raise warnings about null values, set to True to avoid
     """
```
Comment (rportilla-databricks) [inline python/tempo/tsdf.py:322]: Let's add a comment that we will collect stats when the WARNING level is enabled also.

**Review by rportilla-databricks — CHANGES_REQUESTED** [review]

File: python/tempo/tsdf.py (hunk 1 above)
Comment (souvik-databricks) [reply python/tempo/tsdf.py:155]: Changed the condition from 'and' to 'or'

File: python/tempo/tsdf.py (hunk 2 above)
Comment (souvik-databricks) [reply python/tempo/tsdf.py:322]: Added the comment as part of a note in the function.

**Review by rportilla-databricks — CHANGES_REQUESTED** [review]
Please bump the setup.py pypi version to 0.1.6. Then will merge. Thanks!

**Review by rportilla-databricks — APPROVED** [review]

**Pull request merged**